ui/app/AppLayouts/Chat/views/ChatView.qml

```diff
@@ -160,17 +160,33 @@ StatusSectionLayout {
             return false
         }
 
-        let chatContentModule = root.rootStore.currentChatContentModule()
-        if (!chatContentModule) {
+        if (!root.chatContentModule) {
             return false
         }
         // Check if user list is available as an option for particular chat content module
-        return chatContentModule.chatDetails.isUsersListAvailable
+        return root.chatContentModule.chatDetails.isUsersListAvailable
     }
 
     rightPanel: Component {
         id: userListComponent
         UserListPanel {
+            id: userListPanel
+            readonly property bool isFullCommunityList: !root.chatContentModule || !root.chatContentModule.chatDetails || !root.chatContentModule.chatDetails.requiresPermissions
+            property bool wasFullCommunityList: false
+
+            Connections {
+                target: root
+                onChatContentModuleChanged: {
+                    if (!root.chatContentModule ||
+                        (userListPanel.usersModel != null && root.chatContentModule.chatDetails.belongsToCommunity && userListPanel.wasFullCommunityList && userListPanel.isFullCommunityList)) {
+                        // If the previous channel had the full community list already, and we still need the whole list, just keep the old model
```

Comment:
I'm wondering if this isn't a bit dangerous..What if we delete the previous channel and the model is still cached here? I think it would lead to a crash since there are cases where the `usersModel` won't have a binding anymore on the proper `chatContentModule`, but the model reference assigned below.

Reply:
I tested it and it didn't crash, but I totally get your worry. I also think it can be dangerous. 
I'm not sure how to make it safe though (apart from just removing this).

I know back in the day when we switched from one UserList for the whole community to having a specific lists per channel, I suggested having one model for all the members of the community that we can show for unpermissioned channels, and then permissioned channels could have their own model, but I know it wasn't done because it was too complex with the current architecture.

I'll try to figure out a way to do it now.
It would free a lot of memory and also fix the worry that it might crash.
I'll open a new issue for it though and work on it in another PR on top

Reply:
Issue opened here https://github.com/status-im/status-desktop/issues/16288

Reply:
PR opened here: https://github.com/status-im/status-desktop/pull/16301

We can wait until that new PR is merged before merging this one, that way we don't have to merge the possibly dangerous code.

Reply:
Alright, I removed the dangerous code. I was tired of rebasing lal my PRs on this. I want to merge it hahaha